libcfnet/net.c

```diff
@@ -133,9 +133,25 @@ int ReceiveTransaction(const ConnectionInfo *conn_info, char *buffer, int *more)
     {
     case CF_PROTOCOL_CLASSIC:
         ret = RecvSocketStream(conn_info->sd, proto, CF_INBAND_OFFSET);
+        if (ret == SOCKET_RECV_TIMEOUT)
+        {
+            /* We are experiencing problems with connection timeouts. 
+             * This might lead to packages being not delivered in correct
+             * order and unexpected issues like directories being replaced
+             * with files. 
+             * In order to make sure that file transfer is reliable we have to
+             * close connection to avoid broken packages being received. */
+            conn_info->is_timeouted = true;
+        }
         break;
     case CF_PROTOCOL_TLS:
         ret = TLSRecv(conn_info->ssl, proto, CF_INBAND_OFFSET);
```

Comment:
Shouldn't **any** failure of `TLSRecv()` be propagated upwards, and cause the promise to fail?